src/core/pointcloud/qgspointcloudrenderer.h

```diff
@@ -675,6 +675,33 @@ class CORE_EXPORT QgsPointCloudRenderer
      */
     virtual QStringList legendRuleKeys() const;
 
+    /**
+     * Set whether the renderer should also render file labels inside extent
+     * \since QGIS 3.42
+     */
+    void setShowLabels( const bool show ) { mShowLabels = show; }
+
+    /**
+     * Returns whether the renderer shows file labels inside the extent
+     * rectangle
+     * \since QGIS 3.42
+     */
+    bool showLabels() const { return mShowLabels; }
+
+    /**
+       * Sets the text format renderers should use for rendering labels
+       * \note This class takes ownership of textFormat
+       * \since QGIS 3.42
+       */
+    void setLabelTextFormat( QgsTextFormat *textFormat SIP_TRANSFER ) { mLabelTextFormat.reset( textFormat ); }
```

Comment:
QgsTextFormat is implicitly shared -- it's cheap to make copies of this, so should always be passed/stored by value:

```suggestion
    void setLabelTextFormat( const QgsTextFormat & format ) { mLabelTextFormat = format; }
```